front-end/src/app/reports/f3x/submission-workflow/submit-f3x-step2.component.ts

```diff
@@ -0,0 +1,192 @@
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Observable, Subject, takeUntil } from 'rxjs';
+import { Store } from '@ngrx/store';
+import { ConfirmationService, MessageService } from 'primeng/api';
+import { refreshCommitteeAccountDetailsAction } from '../../../store/committee-account.actions';
+import { selectCommitteeAccount } from 'app/store/committee-account.selectors';
+import { LabelUtils, PrimeOptions, StatesCodeLabels, CountryCodeLabels } from 'app/shared/utils/label.utils';
+import { ValidateService } from 'app/shared/services/validate.service';
+import { schema as f3xSchema } from 'fecfile-validate/fecfile_validate_js/dist/F3X';
+import { F3xSummary } from 'app/shared/models/f3x-summary.model';
+import { F3xSummaryService } from 'app/shared/services/f3x-summary.service';
+import { CommitteeAccount } from 'app/shared/models/committee-account.model';
+import { ReportCodeLabelList } from '../../../shared/utils/reportCodeLabels.utils';
+import { updateLabelLookupAction } from '../../../store/label-lookup.actions';
+import { selectReportCodeLabelList } from 'app/store/label-lookup.selectors';
+import { f3xReportCodeDetailedLabels } from '../../../shared/utils/label.utils';
+
+@Component({
+  selector: 'app-submit-f3x-step2',
+  templateUrl: './submit-f3x-step2.component.html',
+})
+export class SubmitF3xStep2Component implements OnInit, OnDestroy {
+  formProperties: string[] = [
+    'treasurer_first_name',
+    'treasurer_last_name',
+    'treasurer_middle_name',
+    'treasurer_prefix',
+    'treasurer_suffix',
+    'filing_password',
+    'truth_statement',
+  ];
+  report: F3xSummary | undefined;
+  report_code = '';
+  stateOptions: PrimeOptions = [];
+  countryOptions: PrimeOptions = [];
+  formSubmitted = false;
+  destroy$: Subject<boolean> = new Subject<boolean>();
+  committeeAccount$: Observable<CommitteeAccount> = this.store.select(selectCommitteeAccount);
+  reportCodeLabelList$: Observable<ReportCodeLabelList> = new Observable<ReportCodeLabelList>();
+  form: FormGroup = this.fb.group(this.validateService.getFormGroupFields(this.formProperties));
+  f3xReportCodeDetailedLabels = f3xReportCodeDetailedLabels;
+  loading: 0 | 1 | 2 = 0;
+
+  constructor(
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private f3xSummaryService: F3xSummaryService,
+    private validateService: ValidateService,
+    private fb: FormBuilder,
+    private store: Store,
+    private messageService: MessageService,
+    protected confirmationService: ConfirmationService
+  ) {}
+
+  ngOnInit(): void {
+    // Refresh committee account details whenever page loads
+    this.store.dispatch(refreshCommitteeAccountDetailsAction());
+    this.stateOptions = LabelUtils.getPrimeOptions(StatesCodeLabels);
+    this.countryOptions = LabelUtils.getPrimeOptions(CountryCodeLabels);
+
+    this.report = this.activatedRoute.snapshot.data['report'];
+    if (this.report?.report_code) {
```

Comment:
`this.report_code = this.report?.report_code || ''` should accomplish the same goal.

Reply:
Great idea